### ISSUE TITLE
Automatic imports

### ISSUE THREAD

**Issue by nullbio** [avh4/elm-format#552]
I'd really like to see automatic imports either built into elm-format, or as a stand-alone tool. This means removing unused imports and adding forgotten imports.

On that note, how does everyone else manage imports? Coming from Go, I'm finding using Elm extremely irritating for this particular reason only. For example, I went to refactor a large file into multiple files, and then I had to painstakingly go through each import one by one, adjust, save, read compiler error, adjust, save, read compiler error, etc. to fix the imports across the new and old files. Surely there's a better way to do this? 

And on top of that, unless you delete all of your imports and re-import them all manually every time you refactor like this, it's quite easy to leave stale imports behind and you have no way of knowing. Does that impact binary size?

**Comment by avh4** [comment]
> I'd really like to see automatic imports either built into elm-format, or as a stand-alone tool. This means removing unused imports and adding forgotten imports.

I agree!  But I don't think elm-format can do it, since it is intended to be run on save for most people, so removing unused imports would mean that if you wrote an import and saved before using it, it would be removed.  Also, elm-format is designed to be able to always give the same output for a given input file without having to know anything about other files that may or may not exist, and I don't think we could add missing imports and still meet that goal.

For your more general questions, you might want to ask at https://discourse.elm-lang.org/ or on the Elm slack to get replies from other Elm users.
